src/Symfony/Component/Validator/Constraints/ImageValidator.php

```diff
@@ -43,6 +44,22 @@ public function validate($value, Constraint $constraint)
             return;
         }
 
+        if ($constraint->strict) {
+            if (!function_exists('imagecreatefromstring')) {
+                throw new RuntimeException('Strict image validation requires installed and enabled GD extension');
```

Comment:
Don't know, but is constraints can be cached and serialized? If yes and constructor wouldn't called then we will miss this check sometimes ang get fatal error in constraint validator.